formats/tree/TreeResultPrinter.php

```diff
@@ -0,0 +1,370 @@
+<?php
+
+namespace SRF\Formats\Tree;
+
+/**
+ * File holding the Tree class.
+ * @author Stephan Gambke
+ */
+use Cdb\Exception;
+use Html;
+use SMW\DIProperty;
+use SMW\DIWikiPage;
+use SMW\ListResultPrinter;
+use SMWQueryResult;
+use Title;
+
+/**
+ * Result printer that prints query results as a tree (nested html lists).
+ *
+ * The available formats are 'tree', 'ultree', 'oltree'. 'tree' is an alias of
+ * 'ultree'. In an #ask query the parameter 'parent' must be set to contain the
+ * name of the property, that gives the parent page of the subject page.
+ *
+ */
+class TreeResultPrinter extends ListResultPrinter {
+
+	private $standardTemplateParameters;
+
+	/**
+	 * @var SMWQueryResult | null
+	 */
+	private $queryResult = null;
+
+	/**
+	 * (non-PHPdoc)
+	 * @see SMWResultPrinter::getName()
+	 */
+	public function getName() {
+		// Give grep a chance to find the usages:
+		// srf-printername-tree, srf-printername-ultree, srf-printername-oltree
+		return \Message::newFromKey( 'srf-printername-' . $this->mFormat )->text();
+	}
+
+	/**
+	 * @return SMWQueryResult
+	 * @throws Exception
+	 */
+	public function getQueryResult() {
+
+		if ( $this->queryResult === null ) {
+			throw new Exception( __METHOD__ . ' called outside of ' . __CLASS__ . '::getResultText().');
```

Comment:
True. The whole design is rather awkward. It should not be necessary to have this check. Maybe at some point I'll introduce a `ResultProcessor` that is instantiated in `getResultText` and destroyed when the method is done.